packages/binding-mqtt/src/mqtt-client.ts

```diff
@@ -161,6 +161,13 @@ export default class MqttClient implements ProtocolClient {
         }
     }
 
+    /**
+     * @inheritdoc
+     */
+    public async requestThingDescription(uri: string): Promise<Content> {
+        throw new Error("Method not implemented");
```

Comment:
Technically, this can be implemented since an MQTT Thing exposed by node-wot publishes its TD with retain. See https://github.com/eclipse-thingweb/node-wot/blob/master/packages/binding-mqtt/src/mqtt-broker-server.ts#L141 It is not standardized afaik but I do not know what speaks against doing a subscription, getting a TD and unsubscribing. Of course it can be done in another PR, I leave it up to you @JKRhb

Reply:
Oh, yeah, very good point, I implemented something similar in `dart_wot` so I can confirm that this is working :) Maybe I would open a follow-up PR for this feature, though, since it might require some discussion regarding the URL-to-topic mapping 🤔

Reply:
Yup sure. Interestingly, this is also mandated by sparkplug with the birth messages (same as expose here). More reading is possible at https://sparkplug.eclipse.org/specification/version/3.0/documents/sparkplug-specification-3.0.0.pdf at page 21. Also, the newer version of it allows any payload to be published there (wink wink TD). 

In the end, as long as the developer provides the URL to fetch from, I think it is fine and no need for big discussions.

Reply:
Hmm, having another look at the MQTT client implementation after @relu91's comment I noticed that it actually does not even implement the `readResource` method where I think a similar logic would be used. So my proposal would be to implement both methods at the same time to arrive at a uniform implementation and keep the discussion focused – would that be okay for you, @relu91? Otherwise, adding the implementation here would also not be a problem, though :)